src/components/views/globals/NewVersionBar.js

```diff
@@ -31,35 +32,56 @@ function checkVersion(ver) {
 
 export default function NewVersionBar(props) {
     const onChangelogClicked = () => {
-        const ChangelogDialog = sdk.getComponent('dialogs.ChangelogDialog');
-
-        Modal.createDialog(ChangelogDialog, {
-            version: props.version,
-            newVersion: props.newVersion,
-            onFinished: (update) => {
-                if(update) {
-                    window.location.reload();
+        if (props.releaseNotes) {
+            const QuestionDialog = sdk.getComponent('dialogs.QuestionDialog');
+            Modal.createDialog(QuestionDialog, {
+                title: "What's New",
+                description: <pre className="changelog_text">{props.releaseNotes}</pre>,
+                button: "Update",
+                onFinished: (update) => {
+                    if(update && PlatformPeg.get()) {
+                        PlatformPeg.get().installUpdate();
+                    }
+                }
+            });
+        } else {
+            const ChangelogDialog = sdk.getComponent('dialogs.ChangelogDialog');
+            Modal.createDialog(ChangelogDialog, {
+                version: props.version,
+                newVersion: props.newVersion,
+                releaseNotes: releaseNotes,
+                onFinished: (update) => {
+                    if(update && PlatformPeg.get()) {
+                        PlatformPeg.get().installUpdate();
+                    }
                 }
-            }
-        });
+            });
+        }
+    };
+
+    const onUpdateClicked = () => {
+        PlatformPeg.get().installUpdate();
     };
 
-    let changelog_button;
-    if (checkVersion(props.version) && checkVersion(props.newVersion)) {
-        changelog_button = <button className="mx_MatrixToolbar_action" onClick={onChangelogClicked}>Changelog</button>;
+    let action_button;
+    if (props.releaseNotes || (checkVersion(props.version) && checkVersion(props.newVersion))) {
```

Comment:
I could do with a comment here explaining that we show a questiondialog if we have releasenotes, or a changelogdialog if we don't.